### PR TITLE
Two compilation fixes

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by keszybz** [comment]
@myrkr, if you could test that this works for you.

**Comment by mrc0mmand** [comment]
Regarding the second commit - maybe it would be for the best to run the minimal supported clang version + two latest ones in the CI instead of the latest three, as we already do for gcc.

**Review by bluca — APPROVED** [review]

**Comment by keszybz** [comment]
> Regarding the second commit - maybe it would be for the best to run the minimal supported clang version + two latest ones in the CI instead of the latest three, as we already do for gcc.

Yeah, that'd make sense.

**Comment by keszybz** [comment]
In the bug:
> Applying 121932189b47c62b4fb91ab3cb1650e4e17b6ad6 fixes the build problem with clang-11.

**Comment by keszybz** [comment]
Oh, I force pushed before creating the PR, but I guess @myrkr took the commit hash from the initial notification. But this shouldn't matter, the change is not important for that test.

File: src/test/test-bitfield.c [hunk 1]
```diff
@@ -213,6 +213,8 @@ TEST(bits) {
         assert_se(_u64 == u64);
 
         /* Verify these use cases are constant-folded. */
+#if !defined(__clang__) || (__clang_major__ >= 13)
```
Comment (yuwata) [inline src/test/test-bitfield.c:216]: Why not `>= 12` ? Or Clang 12 also does not like the tests?

Comment (myrkr) [reply src/test/test-bitfield.c:216]: Here are some links to the compiler explorer:

- clang-13 (compiles): https://godbolt.org/z/n4xveaxG8
- clang-12 (fails): https://godbolt.org/z/Kn8jnqx3x

So it seems that clang version 13 is the minimum version.

Comment (yuwata) [reply src/test/test-bitfield.c:216]: Then, we should update the comment below.

Comment (yuwata) [reply src/test/test-bitfield.c:216]: -> #28601

**Pull request merged**